File: example/web/zHNativeChannel/handlers/PingMsgHandler.ts
import { ChannelBaseHandler } from '../ChannelBaseHandler';
import { ChannelBaseMsg } from '../ChannelBaseMsg';

// PlatformChannelHandler("PingMsg")
export class PingMsgHandler implements ChannelBaseHandler {
  handle(message: ChannelBaseMsg): ChannelBaseMsg {
    return message;
  }
}

File: example/ios/Runner/zHNativeChannel/handler_ios/PingMsgHandler.swift
import Foundation
import zh_native_channel

// PlatformChannelHandler("PingMsg")
final class PingMsgHandler: ChannelBaseHandler {
  func handle(_ message: ChannelBaseMsg) async throws -> ChannelBaseMsg {
    guard let ping = message as? PingMsg else {
      throw ChannelMsgError.typeMismatch(expected: "PingMsg")
    }
    return PingMsg(text: "ios:\(ping.text)")
  }
}
